src/components/MAPI/workernodes/patches.ts

```diff
@@ -0,0 +1,32 @@
+import { NodePool, ProviderNodePool } from 'MAPI/types';
+import * as capiexpv1alpha3 from 'model/services/mapi/capiv1alpha3/exp';
+
+export type NodePoolPatch = (
```

Comment:
Tried to make the node pool creation code a bit lighter by having all feature-specific code in separate components (e.g. a spot instances component is in charge of setting up spot instances). By using patches, we make sure that we don't randomly mutate the node pool data structures from feature-specific components.

Stuff changes in feature components, they return patches to the parent node pool creation form, which applies those patches onto the node pool data structures stored in memory.

So far this pattern has been working pretty well. If I like how it evolves, I'll also use it for cluster creation, and maybe refactor cluster updates/node pool updates to use it, as well.